src/components/Settings/integrations/Mailchimp/MailchimpAccordion.test.tsx

```diff
@@ -149,6 +149,48 @@ describe('MailchimpAccount', () => {
     beforeEach(() => {
       mailchimpAccount = { ...standardMailchimpAccount };
     });
+    it('is connected but with an API key error', async () => {
+      mailchimpAccount.validateKey = false;
+      const mutationSpy = jest.fn();
+      const { queryByText, getByRole } = render(
+        <Components>
+          <GqlMockedProvider<{
+            MailchimpAccount: MailchimpAccountQuery | undefined;
+          }>
+            mocks={{
+              MailchimpAccount: {
+                mailchimpAccount: [mailchimpAccount],
+              },
+            }}
+            onCall={mutationSpy}
+          >
+            <MailchimpAccordion
+              handleAccordionChange={handleAccordionChange}
+              expandedAccordion={IntegrationAccordion.Mailchimp}
+            />
+          </GqlMockedProvider>
+        </Components>,
```

Comment:
If you would be willing, would you be able to make the codebase better by removeing a lot of the repeat code I added?

You can make the  `<Components />` takes props so each test doesn't have to repeat the `GqlMockedProvider` and `MailchimpAccordion` components.